### PR TITLE
Fix ChatChannelVC crash for DM channels

### PR DESCRIPTION
**Pull request by b-onc**
### 🔗 Issue Links

#2091 
https://getstream.slack.com/archives/C01FT5767AS/p1655306508062099

### 🎯 Goal

To prevent the crash

### 📝 Summary

`ChatChannelVC` has this line in `setUp` func:
```swift
messageComposerVC.channelController = client.channelController(for: channelController.cid!)
```
SDK assumes that at this point, `channelController` will have a valid `cid`, but there's no code path to guarantee `synchronize` to be called before this line. So, if a controller doesn't have a valid `cid`, (which is the case if it's been created with `createDirectMessageChannelWith` factory), and `synchronize` hasn't been called on it yet.

### 🛠 Implementation

`ChannelVC` already calls `synchronize` on the controller, but only after this line. So we should only assign a new controller to composer when we're sure we have the valid `cid`

### 🧪 Manual Testing Notes

Use this patch file in the DemoApp and run the app:
[ChatChannelVC_Crash_Reproduce.patch.zip](https://github.com/GetStream/stream-chat-swift/files/8926861/ChatChannelVC_Crash_Reproduce.patch.zip)

### ☑️ Contributor Checklist
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Changelog is updated with client-facing changes
- [x] New code is covered by unit tests
- [x] This change follows zero ⚠️ policy (required)
- [x] Comparison screenshots added for visual changes
- [x] Affected documentation updated (docusaurus, tutorial, CMS)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 - Fix last message of hidden channel with `clearHistory` visible in channel list [#2056](https://github.com/GetStream/stream-chat-swift/issues/2056)
 - Message action title now supports displaying 2 lines of text instead of 1 [#2082](https://github.com/GetStream/stream-chat-swift/pull/2082)
 - Fix Logger persisting config after usage, preventing changing parameters (such as LogLevel) [#2081](https://github.com/GetStream/stream-chat-swift/issues/2081)
+- Fix crash in `ChannelVC` when it's initialized using a `ChannelController` created with `createDirectMessageChannelWith` factory [#2095](https://github.com/GetStream/stream-chat-swift/issues/2095)
```
Comment (evsaev) [inline CHANGELOG.md:15]:
```suggestion
- Fix crash in `ChannelVC` when it's initialized using a `ChannelController` created with `createDirectMessageChannelWith` factory [#2097](https://github.com/GetStream/stream-chat-swift/issues/2097)
```

Comment (b-onc) [reply CHANGELOG.md]: I really need to fix our `get_next_issue_number` lane :D

File: Sources/StreamChatUI/ChatChannel/ChatChannelVC.swift [hunk 2]
```diff
@@ -71,12 +71,23 @@ open class ChatChannelVC: _ViewController,
...
-
-        messageComposerVC.channelController = client.channelController(for: channelController.cid!)
+        
         messageComposerVC.userSearchController = userSuggestionSearchController
+        
+        func setChannelControllerToComposerIfNeeded(cid: ChannelId?) {
+            guard messageComposerVC.channelController == nil else { return }
+            let composerChannelController = channelController.cid.map { client.channelController(for: $0) }
+            messageComposerVC.channelController = composerChannelController
+        }
```
Comment (evsaev) [inline Sources/StreamChatUI/ChatChannel/ChatChannelVC.swift:81]: I'd make it `private` in `ChatChannelVC` scope to keep the lifecycle methods short

**Review by evsaev — APPROVED** [review]
Looks good!

**Pull request merged**